### DISCUSSION TITLE
[URGENTE] No se puede modificar la configuración de ningun curso.

### DISCUSSION THREAD

**Discussion by tecnicoclima** [chamilo/chamilo-lms#5583] [General] [answered]
Hola a todos,

La configuración de todos los cursos está deshabilitada.

Entro en chamilo como administrador y clico 

Administracion ------ lista de cursos

Abro cualquiera de los cursos voy al final de la pantalla  y clico en configuración del curso (quiero modificar los itinerarios y poner un icono) y como pueden ver aparece inhabilitado y no se puede modificar la configuración y guardar la configuración. (es decir que no funciona).

Les adjunto pantallazo

El resto de la plataforma aparentemente funciona bien.

Estoy usando la versión 1.11.26

Y anteriormente siempre ha funcionado, parece como si hay una opción que está deshabilitada y que hasta que no se habilite no se podrá modificar ninguna configuración de ningún curso. (Los cursos anteriores a este error mantienen la configuración pero tampoco se pueden modificar)

No soy capaz de encontrar esta opción y he mirado tutoriales y otros pero no la encuentro.

Estoy pensando en reinstalar la versión, pero creo que no será la solución por que no parece un error si no una mala configuración.

Por eso pregunto si a alguien le ha pasado y ha encontrado la solución.

Muchas gracias

![Captura de pantalla 2024-06-09 100025](https://github.com/chamilo/chamilo-lms/assets/171746114/8df94bfe-787b-48b5-9cd2-048d537470a7)
![Captura de pantalla 2024-06-09 205356](https://github.com/chamilo/chamilo-lms/assets/171746114/e96f8b9d-e8c3-4671-a869-6dd7e14e51b1)

**Comment by NicoDucou** [comment, marked as answer]
Hola @tecnicoclima 

Se puede ver en el codigo la parte siguiente que bloquea el formulario :
```
if (is_settings_editable()) {
    $form->addButtonSave(get_lang('SaveSettings'), 'submit_save');
} else {
    // Is it allowed to edit the course settings?
    if (!is_settings_editable()) {
        $disabled_output = "disabled";
    }
    $form->freeze();
}
```
Y la funcion is_settings_editable esta asi :
```
function is_settings_editable()
{
    return isset($GLOBALS['course_info_is_editable']) && $GLOBALS['course_info_is_editable'];
}
```
El parametro course_info_is_editable esta definido en un archivo de configuracion.
En app/config/course_info.conf.php deberias de tener una linea  con : 
`$course_info_is_editable` = true;`

Tal vez en la actualizacion borraste este archivo y entonces si ne existe el archivo no existe la variable y entonces el curso no esta editable.

Si no existe el archivo deberias de tener a la el archivo por defecto app/config/course_info.conf.dist.php que puedes duplicar en app/config/course_info.conf.php

Saludos

**Comment by tecnicoclima** [comment]
Muchas gracias,

Ya funciona correctamente

Un saludo
